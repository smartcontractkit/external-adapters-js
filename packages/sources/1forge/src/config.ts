import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = '1FORGE'

export const DEFAULT_ENDPOINT = 'forex'
export const DEFAULT_BASE_URL = 'https://api.1forge.com/'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix, true)
  config.api = {
    ...config.api,
    baseURL: config.api.baseURL || DEFAULT_BASE_URL,
    params: {
      api_key: config.apiKey,
    },
  }
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
