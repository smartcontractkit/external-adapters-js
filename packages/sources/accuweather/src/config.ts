import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'ACCUWEATHER'
export const DEFAULT_BASE_URL = 'http://api.accuweather.com/'
export const DEV_BASE_URL = 'http://apidev.accuweather.com/'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL

  return config
}
