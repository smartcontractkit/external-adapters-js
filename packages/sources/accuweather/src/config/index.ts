import { Requester } from '@chainlink/ea-bootstrap'
import { DefaultConfig } from '@chainlink/ea-bootstrap'

export const NAME = 'ACCUWEATHER'
export const DEFAULT_BASE_URL = 'http://api.accuweather.com/'
export const DEV_BASE_URL = 'http://apidev.accuweather.com/'

export const makeConfig = (prefix?: string): DefaultConfig => {
  const config = Requester.getDefaultConfig(prefix, true)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL

  return config
}
