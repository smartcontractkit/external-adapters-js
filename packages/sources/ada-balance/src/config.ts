import { Requester, util } from '@chainlink/ea-bootstrap'
import { Config as BaseConfig } from '@chainlink/types'

export const NAME = 'ADA_BALANCE'

export const DEFAULT_ENDPOINT = 'balance'
export const DEFAULT_PORT = 1337

export interface ExtendedConfig extends BaseConfig {
  httpOgmiosURL?: string
  wsOgmiosURL?: string
  isTLSEnabled?: boolean
  host?: string
  port: number
}

export const makeConfig = (prefix?: string): ExtendedConfig => {
  const baseConfig = Requester.getDefaultConfig(prefix)
  baseConfig.defaultEndpoint = DEFAULT_ENDPOINT
  const rpcPort = util.getEnv('RPC_PORT', prefix)
  return {
    ...baseConfig,
    httpOgmiosURL: util.getEnv('HTTP_OGMIOS_URL', prefix),
    wsOgmiosURL: util.getEnv('WS_OGMIOS_URL', prefix),
    host: util.getEnv('WS_API_ENDPOINT', prefix),
    port: rpcPort ? parseInt(rpcPort) : DEFAULT_PORT,
    isTLSEnabled: !!util.getEnv('IS_TLS_ENABLED', prefix),
  }
}
