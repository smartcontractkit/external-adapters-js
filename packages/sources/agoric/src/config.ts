import { Requester } from '@chainlink/external-adapter'
import { Config } from '@chainlink/types'
import { util } from '@chainlink/ea-bootstrap'

export const DEFAULT_API_ENDPOINT = 'http://localhost:8000/api/oracle'

// This environment variable is needed for the Hack the Orb oracle
// instructions to remain correct.
const LEGACY_API_ENDPOINT_ENV = 'AG_SOLO_ORACLE_URL'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL =
    config.api.baseURL || util.getEnv(LEGACY_API_ENDPOINT_ENV) || DEFAULT_API_ENDPOINT
  config.apiKey = config.apiKey || 'not required'
  return config
}
