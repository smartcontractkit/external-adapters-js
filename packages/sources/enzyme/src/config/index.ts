import { Requester, util } from '@chainlink/ea-bootstrap'
import { Config as BaseConfig } from '@chainlink/ea-bootstrap'
import { ethers } from 'ethers'

export const NAME = 'ENZYME'

export const ENV_ETHEREUM_RPC_URL = 'ETHEREUM_RPC_URL'
export const ENV_FALLBACK_RPC_URL = 'RPC_URL'
export const ENV_ETHEREUM_CHAIN_ID = 'ETHEREUM_CHAIN_ID'
export const ENV_FALLBACK_CHAIN_ID = 'CHAIN_ID'
export const DEFAULT_CHAIN_ID = '1'

export const DEFAULT_ENDPOINT = 'calcNav'

export type Config = BaseConfig & {
  provider: ethers.providers.Provider
}

export const makeConfig = (prefix?: string): Config => {
  const rpcURL = util.getRequiredEnvWithFallback(
    ENV_ETHEREUM_RPC_URL,
    [ENV_FALLBACK_RPC_URL],
    prefix,
  )

  const chainId =
    parseInt(
      util.getEnvWithFallback(ENV_ETHEREUM_CHAIN_ID, [ENV_FALLBACK_CHAIN_ID]) || DEFAULT_CHAIN_ID,
    ) || util.getEnvWithFallback(ENV_ETHEREUM_CHAIN_ID, [ENV_FALLBACK_CHAIN_ID])

  return {
    ...Requester.getDefaultConfig(prefix),
    defaultEndpoint: DEFAULT_ENDPOINT,
    provider: new ethers.providers.JsonRpcProvider(rpcURL, chainId),
  }
}
