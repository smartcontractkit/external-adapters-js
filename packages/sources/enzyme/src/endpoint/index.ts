export * as calcGav from './calcGav'
export * as calcNav from './calcNav'
export * as calcNetValueForSharesHolder from './calcNetValueForSharesHolder'
