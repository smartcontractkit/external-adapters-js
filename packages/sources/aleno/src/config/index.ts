import { AdapterConfig } from '@chainlink/external-adapter-framework/config'

export const config = new AdapterConfig({
  API_KEY: {
    description: 'An API key for Aleno',
    type: 'string',
    required: true,
    sensitive: true,
  },
  WS_API_ENDPOINT: {
    description: 'WS endpoint for Aleno',
    type: 'string',
    default: 'https://state-price-socket.aleno.ai',
  },

  BACKGROUND_EXECUTE_MS: {
    description:
      'The amount of time the background execute should sleep before performing the next request',
    type: 'number',
    default: 10_000,
  },
})
