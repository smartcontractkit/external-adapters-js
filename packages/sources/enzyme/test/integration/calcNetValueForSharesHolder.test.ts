import { AdapterRequest } from '@chainlink/ea-bootstrap'
import { server as startServer } from '../../src'
import { mockEthereumResponseSuccess } from './fixtures'
import { ENV_ETHEREUM_RPC_URL } from '../../src/config'
import { setupExternalAdapterTest } from '@chainlink/ea-test-helpers'
import type { SuiteContext } from '@chainlink/ea-test-helpers'
import { SuperTest, Test } from 'supertest'

const context: SuiteContext = {
  req: null,
  server: startServer,
}

const envVariables = {
  CACHE_ENABLED: 'false',
  [ENV_ETHEREUM_RPC_URL]: process.env[ENV_ETHEREUM_RPC_URL] || 'http://localhost:8545/',
  API_VERBOSE: 'true',
  CHAIN_ID: process.env['CHAIN_ID'] || '42',
}

setupExternalAdapterTest(envVariables, context)

describe('execute', () => {
  const id = '1'

  describe('with calculatorContract/vaultProxy/sharesHolder', () => {
    const data: AdapterRequest = {
      id,
      data: {
        endpoint: 'calcNetValueForSharesHolder',
        calculatorContract: '0x0b2cBB1974f17700531439E3e4AfF5e5D2AADD4A',
        vaultProxy: '0x399acf6102c466a3e4c5f94cd00fc1bfb071d3c1',
        sharesHolder: '0x31d675bd2bdfdd3e332311bef7cb6ba357a5d4e3',
      },
    }

    it('should return success', async () => {
      mockEthereumResponseSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
