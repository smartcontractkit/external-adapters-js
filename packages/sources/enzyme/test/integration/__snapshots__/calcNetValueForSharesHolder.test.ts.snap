// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute with calculatorContract/vaultProxy/sharesHolder should return success 1`] = `
Object {
  "data": Object {
    "netValue": "10000000000000000000",
    "result": "10000000000000000000",
  },
  "jobRunID": "1",
  "result": "10000000000000000000",
  "statusCode": 200,
}
`;
